app/models/user.rb

```diff
@@ -3,13 +3,34 @@ class User < ApplicationRecord
 
   validates :name, presence: true, uniqueness: true
   validates :email, uniqueness: true, format: {with: URI::MailTo::EMAIL_REGEXP}
+  validates :email, uniqueness: true
   validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }
 
+  after_destroy :ensure_an_admin_remains
+  after_create_commit :welcome_mail
+  before_update :ensure_not_admin
+  before_destroy :ensure_not_admin
+
   class Error < StandardError
   end
 
   private
 
+  def admin?
+    email == ADMIN_EMAIL
+  end
+  
+  def ensure_not_admin
+    if admin?
+      errors.add :base, 'Cannot Update or Delete Admin User'
+      throw :abort
+    end
+  end
+
+  def welcome_mail
+    UserMailer.registered(self).deliver_later
```

Comment:
What if i used `deliver_now` here. What is the difference?
Please mention here

Reply:
The main difference is that ``` deliver now ``` delivers the mail immediately where as ```deliver_later``` enques the job , so here we can use ```deliver_later``` as we are not doing any authentication operation.